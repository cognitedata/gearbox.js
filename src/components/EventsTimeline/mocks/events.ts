import { fakeEvents } from '../../../mocks';
import {
  CogniteEventForTimeline,
  EventTimelineType,
  EventTimelineView,
  TimelineEvent,
} from '../interfaces';

export const timelineEvents: TimelineEvent[] = [
  {
    id: 8825861064387,
    type: EventTimelineType.continuous,
    view: EventTimelineView.fill,
  },
  {
    id: 1995162693488,
    type: EventTimelineType.discrete,
    view: EventTimelineView.outline,
  },
];

export const getCogniteEventsForTimeline: (
  tlEvents: TimelineEvent[]
) => CogniteEventForTimeline[] = (tlEvents: TimelineEvent[]) =>
  tlEvents.map(({ id, type, view }) => ({
    ...fakeEvents.find(e => e.id === id)!,
    appearance: { view, type },
  }));
