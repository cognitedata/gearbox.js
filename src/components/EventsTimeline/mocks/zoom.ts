export const zoomMock = (zoomEvents: any = {}): any => {
  const self: any = {};

  self.scaleExtent = () => self;
  self.on = jest.fn().mockImplementation((name: string, fn: any) => {
    zoomEvents[name] = fn;

    return self;
  });
  self.apply = jest.fn();

  return self;
};
