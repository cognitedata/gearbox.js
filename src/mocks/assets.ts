// Copyright 2020 Cognite AS
import { Asset } from '@cognite/sdk';

export const ASSET_DATA = {
  id: 4650652196144007,
  name: 'VAL',
  parentId: 6687602007296940,
  description: 'Valhall plattform',
  metadata: {
    SOURCE_DB: 'workmate',
    SOURCE_TABLE: 'wmate_dba.wmt_location',
    WMT_LOCATION_ID: '1004',
    WMT_LOCATION_WORKSTART: '1999-09-01 07:00:00',
    latestUpdateTimeSource: '1552471210000',
  },
  createdTime: 0,
  lastUpdatedTime: 1553076017185,
};

export const fakeAsset: Asset = {
  name: '23-TE-96148',
  id: 5156972057719,
  rootId: 8515799768286580,
  parentId: 8515799768286580,
  description: 'VRD - PH 1STSTG COMP SEAL GAS HTR',
  metadata: {
    ELC_STATUS_ID: '1211',
    RES_ID: '532924',
    SOURCE_DB: 'workmate',
    SOURCE_TABLE: 'wmate_dba.wmt_tag',
    WMT_AREA_ID: '1600',
    WMT_CATEGORY_ID: '1116',
    WMT_CONTRACTOR_ID: '1686',
    WMT_FUNC_CODE_ID: '4564',
    WMT_LOCATION_ID: '1004',
    WMT_PO_ID: '8309',
    WMT_SAFETYCRITICALELEMENT_ID: '1060',
    WMT_SYSTEM_ID: '4440',
    WMT_TAG_CREATED_DATE: '2009-06-26 15:36:37',
    WMT_TAG_CRITICALLINE: 'N',
    WMT_TAG_DESC: 'VRD - PH 1STSTG COMP SEAL GAS HTR',
    WMT_TAG_GLOBALID: '1000000000682252',
    WMT_TAG_HISTORYREQUIRED: 'Y',
    WMT_TAG_ID: '346452',
    WMT_TAG_ID_ANCESTOR: '346633',
    WMT_TAG_ISACTIVE: '1',
    WMT_TAG_ISOWNEDBYPROJECT: '0',
    WMT_TAG_LOOP: '96148',
    WMT_TAG_MAINID: '681760',
    WMT_TAG_NAME: '23-TE-96148',
    WMT_TAG_UPDATED_BY: '8137',
    WMT_TAG_UPDATED_DATE: '2014-07-11 09:25:15',
  },
  createdTime: new Date(0),
  lastUpdatedTime: new Date(0),
};
