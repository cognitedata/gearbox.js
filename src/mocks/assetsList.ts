// Copyright 2020 Cognite AS
import { Asset } from '@cognite/sdk';
import { AdvancedSearch } from '../interfaces';

export const vmateDba = 'wmate_dba.wmt_location';
export const randomTime = '1999-09-01 07:00:00';
export const randomTimeDateObj1 = new Date('2002-02-02 02:02:02');
export const randomTimeDateObj2 = new Date('2002-02-22 22:22:22');
export const valDescription = 'Valhall plattform';
export const gasCompressionString = 'GAS COMPRESSION AND RE-INJECTION (PH)';
export const wmtag = 'wmate_dba.wmt_tag';

export const SKA: Asset = {
  id: 8129784932439587,
  rootId: 8129784932439587,
  lastUpdatedTime: randomTimeDateObj2,
  createdTime: randomTimeDateObj1,
  name: 'SKA',
  description: 'Skarv',
  metadata: {
    SOURCE_DB: 'workmate',
    SOURCE_TABLE: vmateDba,
    WMT_LOCATION_ACTIVE: 'Y',
    WMT_LOCATION_CODE: 'SKA',
    WMT_LOCATION_EXTENDACTIVEWOP: '4',
    WMT_LOCATION_EXTERNALOWNERSHIP: 'Y',
    WMT_LOCATION_ID: '1010',
    WMT_LOCATION_MAITIS: 'Y',
    WMT_LOCATION_NAME: 'Skarv',
    WMT_LOCATION_NOCOPIESDEFAULTIC: '4',
    WMT_LOCATION_NOCOPIESWOPERMIT: '1',
    WMT_LOCATION_OPERATIONHOURS: '12',
    WMT_LOCATION_SIMULATETIMEFRAME: '60',
    WMT_LOCATION_SJAMAXNOOFTASKS: '99',
    WMT_LOCATION_USEPLOTALTITUDE: 'Y',
    WMT_LOCATION_WORKSTART: randomTime,
  },
};

export const IAA: Asset = {
  id: 7793176078609329,
  rootId: 7793176078609329,
  lastUpdatedTime: randomTimeDateObj2,
  createdTime: randomTimeDateObj1,
  name: 'IAA',
  description: 'IAA Root node',
  metadata: {
    ASSETSCOPENAME: 'IAA',
    DESCRIPTION: 'IAA Root node',
    NAME: 'IAA',
    PARENTUID: '',
    SOURCE: 'Valid',
    SOURCEID: 'Valid.dbo.AkerBP_UNION_ALL_TAG',
    SOURCE_DB: 'valid',
    SOURCE_TABLE: 'iaaassethierarchy',
    TYPE: 'AssetHierarchy',
    UID: 'IAA',
  },
};

export const VAL: Asset = {
  id: 3623339785663936,
  rootId: 3623339785663936,
  lastUpdatedTime: randomTimeDateObj2,
  createdTime: randomTimeDateObj1,
  name: 'VAL',
  description: valDescription,
  metadata: {
    SOURCE_DB: 'workmate',
    SOURCE_TABLE: vmateDba,
    WMT_LOCATION_ACTIVE: 'Y',
    WMT_LOCATION_CODE: 'VAL',
    WMT_LOCATION_EXTENDACTIVEWOP: '4',
    WMT_LOCATION_EXTERNALOWNERSHIP: 'Y',
    WMT_LOCATION_ID: '1004',
    WMT_LOCATION_MAITIS: 'Y',
    WMT_LOCATION_NAME: valDescription,
    WMT_LOCATION_NOCOPIESDEFAULTIC: '2',
    WMT_LOCATION_NOCOPIESWOPERMIT: '2',
    WMT_LOCATION_OPERATIONHOURS: '12',
    WMT_LOCATION_PROGVALUE: 'wmt_location_weudefaultlocation',
    WMT_LOCATION_SIMULATETIMEFRAME: '15',
    WMT_LOCATION_SJAMAXNOOFTASKS: '99',
    WMT_LOCATION_USEPLOTALTITUDE: 'Y',
    WMT_LOCATION_WORKSTART: randomTime,
  },
};

export const SearchValue: AdvancedSearch = {
  name: '20-PA-001A',
  description: 'Crude pump',
  metadata: [
    { id: 1, key: 'manifacturer', value: 'FRAMO' },
    { id: 2, key: 'manifacturer2', value: 'TESLA' },
  ],
};

export const assetsList = [SKA, IAA, VAL];
