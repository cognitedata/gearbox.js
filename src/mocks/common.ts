// Copyright 2020 Cognite AS
/**
 * Mock wait for Promise resolve after @param timeout
 *
 * @param ms milliseconds timeout
 *
 * @return resolved Promise
 */

export const sleep = (ms: number): Promise<void> =>
  new Promise(resolve => setTimeout(resolve, ms));

/**
 * Generate random Number with @param (or default: 10) number of digits
 *
 * @param length possible positive digit
 *
 * @return randomized Number
 */

export const generateNumber = (length: number = 10): number => {
  if (length > 15) {
    return 0;
  }

  const aggregator = new Array(length).fill(null);

  return Number(
    aggregator.map(() => Math.floor(Math.random() * 10) || 1).join('')
  );
};
