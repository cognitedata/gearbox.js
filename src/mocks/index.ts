export * from './assets';
export * from './assetsList';
export * from './assetsListV2';
export * from './common';
export * from './datapoints';
export * from './documents';
export * from './events';
export * from './mockSdk';
export * from './threeD';
export * from './timeseriesListV2';
export * from './timeseriesDataPointsList';
export * from './threeDNodesList';
