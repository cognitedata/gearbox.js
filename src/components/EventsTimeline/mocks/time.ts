import { baseTimestamp } from '../../../mocks';

export const start = baseTimestamp;
export const end = baseTimestamp + 60 * 24 * 60 * 60 * 1000;
