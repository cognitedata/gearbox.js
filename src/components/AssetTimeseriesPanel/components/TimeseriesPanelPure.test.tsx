// Copyright 2020 Cognite AS
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import renderer from 'react-test-renderer';
import { timeseriesListV2 } from '../../../mocks';
import {
  TimeseriesPanelPure,
  TimeseriesPanelPureComponent,
} from './TimeseriesPanelPure';

configure({ adapter: new Adapter() });

const timeseries = timeseriesListV2.slice(0, 3);

describe('TimeseriesPanelPure', () => {
  it('Renders without exploding', () => {
    const wrapper = mount(
      <TimeseriesPanelPure assetTimeseries={timeseries} />
    ).find(TimeseriesPanelPureComponent);
    expect(wrapper.find('Styled(Collapse)')).toHaveLength(1);
    expect(wrapper.find('Styled(CollapsePanel)')).toHaveLength(3);

    wrapper
      .find('Styled(CollapsePanel)')
      .at(0)
      .simulate('click');
    expect(wrapper.find('TimeseriesChartMetaPure')).toEqual({});
  });

  it('Should match the snapshot', () => {
    const tree = renderer.create(
      <TimeseriesPanelPure assetTimeseries={timeseries} />
    );
    expect(tree).toMatchSnapshot();
  });

  it('Should render a message if timeseries are empty', () => {
    const wrapper = mount(<TimeseriesPanelPure assetTimeseries={[]} />);
    expect(wrapper.text()).toEqual('No timeseries linked to this asset');
  });

  it('Should render special message if provided and timeseries are empty', () => {
    const wrapper = mount(
      <TimeseriesPanelPure
        assetTimeseries={[]}
        strings={{ noTimeseriesSign: 'nothing found' }}
      />
    );
    expect(wrapper.find(TimeseriesPanelPureComponent).text()).toEqual(
      'nothing found'
    );
  });
});
