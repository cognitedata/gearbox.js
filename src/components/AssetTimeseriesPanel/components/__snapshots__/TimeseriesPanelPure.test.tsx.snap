// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeseriesPanelPure Should match the snapshot 1`] = `
<div
  className="sc-kgoBCf jUmofq"
>
  <div
    className="ant-collapse ant-collapse-icon-position-left sc-kGXeez hdOaOS"
    role={null}
  >
    <div
      className="ant-collapse-item sc-chPdSV cBMJjQ"
    >
      <div
        aria-expanded="false"
        className="ant-collapse-header"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <i
          aria-label="icon: right"
          className="anticon anticon-right ant-collapse-arrow"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="right"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
            />
          </svg>
        </i>
        IA_21PT1019.AlarmByte
      </div>
    </div>
    <div
      className="ant-collapse-item sc-chPdSV cBMJjQ"
    >
      <div
        aria-expanded="false"
        className="ant-collapse-header"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <i
          aria-label="icon: right"
          className="anticon anticon-right ant-collapse-arrow"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="right"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
            />
          </svg>
        </i>
        IA_21PT1019.CommandWord
      </div>
    </div>
    <div
      className="ant-collapse-item sc-chPdSV cBMJjQ"
    >
      <div
        aria-expanded="false"
        className="ant-collapse-header"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <i
          aria-label="icon: right"
          className="anticon anticon-right ant-collapse-arrow"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="right"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
            />
          </svg>
        </i>
        IA_21PT1019.PTAB
      </div>
    </div>
  </div>
</div>
`;
